### PR TITLE
Add skeleton of wgpu-py

### PR DESCRIPTION
**Pull request by Vipitis**
Starting from the top of #1 

This PR starts with a squashed clone of wgpu-py at the 0.13.2 release and then removes all parts that are not required.
~~all inits and config are kept~~
~~no modifications to any files have been made~~

- Is there anything else that needs to go?
- will there be any issues by keeping the config files as is?

### CODE REVIEW

**Comment by Korijn** [comment]
I will have a lot of feedback at this stage. I will post code review comments, but if you refer me to just add commits to the branch directly, tell me and I will.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+* @Korijn
```
Comment (Korijn) [inline .github/CODEOWNERS:1]: You can put either your own name, or remove the `CODEOWNERS` file

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+      - main
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+
+  lint-build:
```
Comment (Korijn) [inline .github/workflows/ci.yml:15]: you can remove all of the following CI jobs:

- test-codegen-build
- test-minimal-import-build
- test-pyinstaller-build
- release-builds
- qemu-release-builds
- sdist-build
- publish

File: CHANGELOG.md
Comment (Korijn) [inline CHANGELOG.md:1]: you can clear the changelog and reset it: https://keepachangelog.com/en/1.1.0/

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,25 @@
+BSD 2-Clause License
+
+Copyright (c) 2019-2023, Almar Klein, Korijn van Golen
```
Comment (Korijn) [inline LICENSE:3]: You can put your own name here

File: examples/shadertoy_blink.py
Comment (Korijn) [inline examples/shadertoy_blink.py:1]: you can remove the `shadertoy_` prefix for all examples

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+]
+build-backend = "setuptools.build_meta"
+
+[tool.cibuildwheel]
+# we only build on one python version since the wheels are not bound to it
+build = "cp39-*"
+
+# we can't list requests under build-system.requires because
+# that step happens _after_ the before-build command
+before-build = "pip install requests && python download-wgpu-native.py"
```
Comment (Korijn) [inline pyproject.toml:14]: not needed

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+per-file-ignores =
+    tests/test_compute.py: F821,F722
+    tests/test_gui_glfw.py: F821,F722
+    tests/test_wgpu_native_basics.py: F821,F722
+    tests/test_wgpu_native_render.py: F821,F722
+    tests/test_wgpu_native_render_tex.py: F821,F722
+    tests/test_wgpu_native_compute_tex.py : F821,F722
+    examples/*.py: F821,F722
+    examples/triangle_qt*.py: E402
```
Comment (Korijn) [inline setup.cfg:24]: most of these files don't exist

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    resources_globs.append("*-release.dll")
+else:
+    pass  # don't include binaries; user will have to arrange for the lib
+
+runtime_deps = ["cffi>=1.15.0", "rubicon-objc>=0.4.1; sys_platform == 'darwin'"]
+extra_deps = {
+    "jupyter": ["jupyter_rfb>=0.4.2"],
+    "glfw": ["glfw>=1.9"],
+    "docs": ["sphinx>7.2", "sphinx_rtd_theme"],
+}
```
Comment (Korijn) [inline setup.py:36]: you can remove lines 15-36

File: setup.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+setup(
+    name=NAME,
+    version=VERSION,
+    packages=find_packages(
+        exclude=["codegen", "codegen.*", "tests", "tests.*", "examples", "examples.*"]
```
Comment (Korijn) [inline setup.py:42]: you can remove codegen exclusions

File: setup.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    package_data={f"{NAME}.resources": resources_globs},
+    python_requires=">=3.8.0",
+    install_requires=runtime_deps,
+    extras_require=extra_deps,
+    license="BSD 2-Clause",
+    description=SUMMARY,
+    long_description=open("README.md").read(),
+    long_description_content_type="text/markdown",
+    author="Almar Klein",
+    author_email="almar.klein@gmail.com",
```
Comment (Korijn) [inline setup.py:53]: you can put your own name here

File: setup.py [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    url="https://github.com/pygfx/wgpu-py",
```
Comment (Korijn) [inline setup.py:54]: url needs to be updated

File: setup.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    cmdclass={"bdist_wheel": bdist_wheel},
+    data_files=[("", ["LICENSE"])],
+    entry_points={
+        "pyinstaller40": [
+            "hook-dirs = wgpu.__pyinstaller:get_hook_dirs",
+            "tests = wgpu.__pyinstaller:get_test_dirs",
+        ],
+    },
```
Comment (Korijn) [inline setup.py:62]: you can remove pyinstaller support

File: wgpu/__init__.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
+"""
+The wgpu library is a Python implementation of WebGPU.
```
Comment (Korijn) [inline wgpu/__init__.py:2]: you can rename the package to `shadertoy` :)

File: wgpu/__init__.py [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+from ._coreutils import logger  # noqa: F401,F403
+from ._diagnostics import diagnostics  # noqa: F401,F403
+from .flags import *  # noqa: F401,F403
+from .enums import *  # noqa: F401,F403
+from .classes import *  # noqa: F401,F403
+from .gui import WgpuCanvasInterface  # noqa: F401,F403
+from . import utils  # noqa: F401,F403
+from . import backends  # noqa: F401,F403
+from . import resources  # noqa: F401,F403
```
Comment (Korijn) [inline wgpu/__init__.py:13]: lots of these imports can go :)

File: wgpu/__init__.py [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+# The API entrypoint, from wgpu.classes - gets replaced when a backend loads.
+gpu = GPU()  # noqa: F405
+
+
+# Temporary stub to help transitioning
+def request_adapter(*args, **kwargs):
+    """Deprecated!"""
+    raise DeprecationWarning(
+        "wgpu.request_adapter() is deprecated! Use wgpu.gpu.request_adapter() instead."
+    )
```
Comment (Korijn) [inline wgpu/__init__.py:29]: all of this can go

File: examples/shadertoy_blink.py
Comment (Vipitis) [reply examples/shadertoy_blink.py:1]: I want to go through all the example and check their source/license in a later PR

**Comment by Korijn** [comment]
It would be a ton of effort for me to capture my feedback in more comments so I just pushed some commits to this branch. Feel free to revert anything if you feel differently.

I used these two guides to put pyproject.toml together:

* https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license
* https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata

**Comment by Korijn** [comment]
With CI passing and at least one example under screenshot testing, I think this is ready to merge.

**Comment by Vipitis** [comment]
Thanks for all the changes. I was reading through nearly the same resources yesterday. 

The only issue I see with calling it just `shadertoy` is that it might be protected and we are clearly not affiliated with the website. So perhaps something like `wgpu-shadertoy` would work instead. 

As for linting, I welcome new capabilities. Something that could be interesting, is that shadercode is just python strings - and could maybe get it's own linting rules (for glsl or wgsl specific).

It looks good to merge to me!

**Comment by Korijn** [comment]
> The only issue I see with calling it just `shadertoy` is that it might be protected and we are clearly not affiliated with the website. So perhaps something like `wgpu-shadertoy` would work instead.

That makes sense. We can rename the package to wgpu_shadertoy?

> As for linting, I welcome new capabilities. Something that could be interesting, is that shadercode is just python strings - and could maybe get it's own linting rules (for glsl or wgsl specific).

Easiest way to accomplish that is probably to put shaders in their own .glsl and .wgsl files.

**Pull request merged**